server.go

```diff
@@ -418,6 +418,35 @@ func (server *Server) listenSignals() {
 	server.Stop()
 }
 
+func createClientTLSConfig(tlsOption *TLS) (*tls.Config, error) {
+	if tlsOption == nil {
+		return nil, errors.New("no TLS provided")
```

Comment:
IIUC, then we only hit this case if no TLS was configured on the entry point.

Could we handle this error further up the stack, e.g., in `loadConfig`?